ui-charts/src/manchette/hooks/useManchetteWithSpaceTimeChart.tsx

```diff
@@ -362,161 +325,345 @@ const useManchetteWithSpaceTimeChart = ({
     setState((prev) => ({ ...prev, zoomMode: !prev.zoomMode }));
   }, []);
 
-  const manchetteProps = useMemo(
-    () => ({
-      contents: waypointsToDisplay,
-      zoomYIn,
-      zoomYOut,
-      resetZoom,
-      toggleMode,
-      yZoom,
-      isProportional,
-      yOffset,
-    }),
-    [waypointsToDisplay, zoomYIn, zoomYOut, resetZoom, toggleMode, yZoom, isProportional, yOffset]
+  const handleXZoom = useCallback(
+    (newXZoom: number, xPosition = (spaceTimeChartRef?.current?.offsetWidth || 0) / 2) => {
+      setState((prev) => ({
+        ...prev,
+        ...zoomX(prev.xZoom, prev.xOffset, newXZoom, xPosition),
+      }));
+    },
+    [spaceTimeChartRef]
   );
 
-  const spaceTimeChartProps = useMemo(
-    () => ({
-      operationalPoints: simplifiedWaypoints,
-      spaceScales: computedScales,
-      timeScale: zoomValueToTimeScale(xZoom),
-      paths,
-      xOffset,
-      yOffset: -yOffset + 14,
-      timeOrigin,
-      spaceOrigin,
-      rect,
-      onZoom: ({ delta, position }: Parameters<NonNullable<SpaceTimeChartProps['onZoom']>>[0]) => {
-        if (isShiftPressed) {
-          handleXZoom(xZoom + delta, position.x);
+  const spaceScales = useMemo(() => {
+    // Here, we first compute the base scales, and then we insert a flat step for each split point:
+    const baseScales = getScales(
+      waypoints,
+      { height, isProportional, yZoom },
+      minZoomMillimeterPerPx,
+      maxZoomMillimeterPerPx
+    );
+
+    if (!splitPoints) return baseScales;
+
+    // Constant scale:
+    if (isProportional) {
+      const baseScale = baseScales[0];
+      const coefficient: number =
+        'coefficient' in baseScale && typeof baseScale.coefficient === 'number'
+          ? baseScale.coefficient
+          : (baseScale.to - baseScale.from) / baseScale.size;
+
+      return splitPoints
+        .flatMap(({ position, size: splitPointHeight }) => [
+          {
+            to: position,
+            coefficient,
+          },
+          {
+            to: position,
+            size: splitPointHeight,
+          },
+        ])
+        .concat({
+          to: baseScales.at(-1)!.to,
+          coefficient,
+        });
+    }
+
+    // Varying scales:
+    const allScales: SpaceScale[] = [];
+    baseScales.forEach((baseScale) => {
+      const coefficient =
+        'coefficient' in baseScale && typeof baseScale.coefficient === 'number'
+          ? baseScale.coefficient
+          : (baseScale.to - baseScale.from) / baseScale.size;
+
+      // Search for split point strictly BETWEEN from and to:
+      const splitPoint = splitPoints.find(
+        (point) => point.position > baseScale.from && point.position < baseScale.to
+      );
```

Comment:
There might be several splitPoints in a single scale

Reply:
Good catch, thanks! I just pushed a fix.